src/spec-node/devContainersSpecCLI.ts

```diff
@@ -388,9 +391,9 @@ async function doBuild({
 			if (buildxPlatform || buildxPush) {
 				throw new ContainerError({ description: '--platform or --push require dockerfilePath.' });
 			}
-			if (argImageName) {
-				await dockerPtyCLI(params, 'tag', updatedImageName, argImageName);
-				imageNameResult = argImageName;
+			if (imageNames) {
+				await Promise.all(imageNames.map(imageName => dockerPtyCLI(params, 'tag', updatedImageName[0], imageName)));
```

Comment:
It is not clear to me why we always want the first element of `updatedImageName`.  Does `extendImage` return an array with a single value in it? If that's the case, can it just return the element itself?

Reply:
>  Does extendImage return an array with a single value in it?

Yes, that's the case

> If that's the case, can it just return the element itself?

If we make `extendImage` return a single element, then things get complex because [buildNamedImageAndExtend](https://github.com/devcontainers/cli/blob/65044246c2d73176004cebdd6c00dd2cd9483896/src/spec-node/singleContainer.ts#L106) (which calls `extendImage` based on a condition) then returns a `string | string[]`. We have to fix all compiler issues and also we aren't sure if we want to to use `updatedImageName` or `updatedImageName[]`.

Ref - https://github.com/devcontainers/cli/pull/61/commits/1b598c36c8282628ae99a16277cba2af603a4894 which made extendImage return [] because tests were failing and that change simplied the compiler checks (commit ref)[https://github.com/devcontainers/cli/pull/61/commits/65044246c2d73176004cebdd6c00dd2cd9483896]